src/main/java/com/example/manage_shops/controller/RoleController.java

```diff
@@ -20,8 +19,39 @@ public RoleController(RoleService roleService, Commons commons) {
         this.commons = commons;
     }
 
-    @GetMapping("/getAll")
-    public List<Role> getAllRole(String roleName) {
-        return roleService.getAllRole(roleName);
+    @CrossOrigin(origins = "http://localhost:8080", maxAge = 36000000L)
+    @GetMapping("/getAll/{roleId}")
+    public ResponseEntity<?> getAllRole(@PathVariable int roleId) throws MyValidateException {
+        return ResponseEntity.ok(roleService.getAllRole(roleId));
+    }
+
+    @CrossOrigin(origins = "http://localhost:8080", maxAge = 36000000L)
+    @GetMapping("/getAll/{roleId}/{roleIdUser}")
+    public ResponseEntity<?> getAllRole(@PathVariable int roleId, @PathVariable int roleIdUser) throws MyValidateException {
+        return ResponseEntity.ok(roleService.getRoleById(roleId, roleIdUser));
+    }
+
+    @CrossOrigin(origins = "http://localhost:8080", maxAge = 36000000L)
+    @PostMapping("/save/{roleIdUser}")
```

Comment:
sao cái này save role lại gửi roleId lên nhể

Reply:
cháu gửi theo role của thg user lên để validate role

Reply:
còn Role thì gửi qua body

Reply:
bỏ gửi role đi nhé. bây giờ nếu user đó có roleId =4, nhưng nó sửa url thành roleId=1. thì đổi được role của user đó r

Reply:
http://localhost:8080/save/1 ví dụ như này. cái số 1 kia hoàn toàn sửa được